next/src/validation/schema.ts

```diff
@@ -135,15 +152,26 @@ export function validateSchema(
   schema: JsfSchema,
   options: ValidationOptions = {},
   path: ValidationErrorPath = [],
-  rootJsonLogicBag?: JsonLogicBag,
+  rootJsonLogicContext?: JsonLogicContext,
 ): ValidationError[] {
-  // If we have a rootJsonLogicBag, we shoud use that. If not, we try to check for the 'x-jsf-logic' property in the schema.
-  let jsonLogicBag = rootJsonLogicBag
-  if (!rootJsonLogicBag && schema['x-jsf-logic']) {
-    jsonLogicBag = {
-      schema: schema['x-jsf-logic'],
+  let jsonLogicContext = rootJsonLogicContext
+  let jsonLogicRootSchema: JsonLogicRootSchema | undefined
+
+  // If we have a root jsonLogicContext, we shoud use that.
+  // If not, it probably means the current schema is the root schema (or that there's no json-logic node in the current schema)
+  if (!rootJsonLogicContext && schema['x-jsf-logic']) {
+    // - We should set the jsonLogicContext's schema as the schema in the 'x-jsf-logic' property
+    const { validations, computedValues, ...rest } = schema['x-jsf-logic']
+    const JsonLogicBagSchema: JsonLogicRules = {
+      validations,
+    }
+    jsonLogicContext = {
+      schema: JsonLogicBagSchema,
```

Comment:
I guess this was a renaming accident? 😬 
```suggestion
    const json_logic_rules: JsonLogicRules = {
      validations,
    }
    jsonLogicContext = {
      schema: json_logic_rules,
```

Reply:
indeed, fixing it 👍